packages/network/src/logic/Node.ts

```diff
@@ -430,8 +404,12 @@ export class Node extends EventEmitter {
             }
         })
 
-        this.seenButNotPropagatedSet.delete(streamMessage)
+        if (subscribers.length === 0) {
+            this.logger.warn('no neighbors to propagate message %o to', streamMessage.getMessageID().toArray())
+        }
+
         this.emit(Event.MESSAGE_PROPAGATED, streamMessage)
```

Comment:
Previously this wouldn't fire for `subscribers.length === 0`, is it correct to still fire `Event.MESSAGE_PROPAGATED` if there were "no neighbors to propagate message to"?

Reply:
Looks like this event is currently not used anywhere. That being said, your suggestion makes sense. I'll fire the event only if there were subscribers. (Could also consider removing the event but maybe leave that for a separate time.)